frontend/src/components/transaction-modal/transaction-modal.tsx

```diff
@@ -0,0 +1,101 @@
+import { formatDate } from '../../lib/date'
+import { BreakText } from '../break-text'
+import { Button } from '../button'
+import { CodeBlock } from '../code-block'
+import { Dialog } from '../dialog'
+import { KeyValueTable } from '../key-value-table'
+import type {
+  ParsedTx,
+  TransactionType
+} from '../transaction-manager/transaction-types'
+
+interface TransactionModalProps {
+  transactions: ParsedTx[]
+  onRespond: (txId: string, decision: boolean) => void
+}
+
+const TRANSACTION_TITLES: {
+  [Type in TransactionType]: string
+} = {
+  unknown: 'Unknown transaction',
+  orderSubmission: 'Order submission',
+  withdrawSubmission: 'Withdraw submission',
+  voteSubmission: 'Vote submission',
+  delegateSubmission: 'Delegate submission',
+  undelegateSubmission: 'Undelegate submission'
```

Comment:
chore (blocking): missing types? e.g. proposal submission.

Reply:
Good catch, these types are another thing that would be nice to make publishable from the monorepo

Reply:
There are actually quite a few missing transaction types. Wondering if rather than typing them all here and now we should just make the transaction payload a generic object. After all, all that we do with that object is render it out in a code block.

Then we can potentially publish these types elsewhere either from vegawallet-service-api-client where some are already added, or perhaps from the monorepo.

Reply:
I've added an enum for all available transaction types which helps make sure we have all the translations covered off as well as tidied up the parseTx function as we can loop over these.